File: f148(orienteering)/C++/orienteering.cpp
#include <iostream>
#include <utility>
#include <vector>
#include <algorithm>

using namespace std;

int main(int argc, char const *argv[]) {
  ios_base::sync_with_stdio(false);
  cin.tie(0);
  int row, col, num_of_targets;
  while (cin >> row >> col >> num_of_targets) {
    vector< pair<char, pair<int, int> > > field;
    char target;
    for (int r = 0; r < row; r++) {
      for (int c = 0; c < col; c++) {
        cin >> target;
        if (target != '0') field.push_back(pair<char, pair<int, int> >(target, pair<int, int>(r, c)));
      }
    }
    if (field.size() < num_of_targets) {
      cout << "Mission fail." << '\n';
      continue;
    }
    sort(field.begin(), field.end());
    for (int t = 0; t < num_of_targets; t++) {
      pair<int, int> target = field.at(t).second;
      cout << target.first << ' ' << target.second << '\n';
    }
  }
  return 0;
}

File: f148(orienteering)/Python/orienteering.py
from sys import stdin
while True:
    line = stdin.readline()
    if not line: break
    row, col = map(int, line.split())
    num_of_targets = int(stdin.readline())
    pos = {}
    for i in range(row):
        col = list(stdin.readline().split())
        for p, t in enumerate(col):
            if t != '0': pos[t] = [i, p]
    if len(pos) < num_of_targets:
        print('Mission fail.')
        continue
    ordered_pos = sorted(pos.items(), key = lambda d: d[0])
    for n in range(num_of_targets):
        print(*ordered_pos[n][1])
